debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java

```diff
@@ -773,6 +776,29 @@ void allSlotsInIntermediateClassDataDo () throws CorruptDataException
 			classWalkerCallback.addSection(clazz, cursor, count.longValue(), "intermediateClassDataSection", true);
 		}
 	}
+
+	void allSlotsInVarHandleMethodTypeLookupTableDo() throws CorruptDataException
+	{
+		if (J9BuildFlags.J9VM_OPT_METHOD_HANDLE && !J9BuildFlags.J9VM_OPT_OPENJDK_METHODHANDLE) {
+			try {
+				int count = romClass.varHandleMethodTypeCount().intValue();
+
+				if (count > 0) {
+					Pointer cursorVoidEA = romClass.varHandleMethodTypeLookupTableEA();
+					VoidPointer cursorVoid = SelfRelativePointer.cast(cursorVoidEA).get();
+					U16Pointer cursor = U16Pointer.cast(cursorVoid);
+
+					classWalkerCallback.addSection(clazz, cursor, count * U16.SIZEOF, "varHandleMethodTypeLookupTable", true);
+					for (int i = 0; i < count; i++) {
+						classWalkerCallback.addSlot(clazz, SlotType.J9_U16, cursor.add(i), "cpIndex");
+					}
+				}
+			} catch (NoSuchFieldException e) {
+				throw new CorruptDataException(e);
```

Comment:
When testing this with Java 23 on a system dump produced by Java 8, I encounter
```
Caused by: com.ibm.j9ddr.CorruptDataException: java.lang.NoSuchFieldException
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.RomClassWalker.allSlotsInVarHandleMethodTypeLookupTableDo(RomClassWalker.java:799)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.RomClassWalker.allSlotsInObjectDo(RomClassWalker.java:164)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.LinearDumper.getAllRegions(LinearDumper.java:324)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.LinearDumper.gatherLayoutInfo(LinearDumper.java:449)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.commands.DumpRomClassLinearCommand.run(DumpRomClassLinearCommand.java:63)
	... 19 more
Caused by: java.lang.NoSuchFieldException
	at com.ibm.j9ddr.vm29.pointer.generated.J9ROMClassPointer.varHandleMethodTypeCount(J9ROMClassPointer.java:740)
	at com.ibm.j9ddr.vm29.tools.ddrinteractive.RomClassWalker.allSlotsInVarHandleMethodTypeLookupTableDo(RomClassWalker.java:784)
	... 23 more
```
This signals a flaw in the generation pattern for DDR code. I will have a fix ready shortly.

Reply:
See https://github.com/eclipse-openj9/openj9/pull/19195. @babsingh Perhaps you could review it?

Reply:
Yes, I will review it.